hazelcast/src/main/java/com/hazelcast/internal/util/ConcurrencyUtil.java

```diff
@@ -47,7 +47,7 @@ public String toString() {
 
     // Default executor for async callbacks: ForkJoinPool.commonPool() or a thread-per-task executor when
     // the common pool does not support parallelism
-    private static Executor defaultAsyncExecutor;
+    private static volatile Executor defaultAsyncExecutor;
```

Comment:
no need for this field to be `volatile`, it is initialized in main thread before other threads are spawned (previously discussed here https://github.com/hazelcast/hazelcast/pull/18181/files#r578964018).